client/cmd/dexc-desktop/gen.go

```diff
@@ -0,0 +1,10 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:generate go run github.com/tc-hib/go-winres@v0.3.1 make --in winres.json --arch "386,amd64"
+
+package main
+
+// After generating the rsrc_windows_amd64.syso file, it will be included in the
+// binary when making an amd64 build for windows.
```

Comment:
`go build` picks up syso files and includes them in the binary.  In this case it's a way to get windows resources into the executable.  This syso file is a special windows resource file.  This is the stuff the Windows ABI understands, giving it e.g. the info in the Version table when you right click a file and see properties in Windows Explorer, give a list of icons the explorer can read and use, etc.

The naming structure `blah_<os>_<arch>` is recognize by `go build` for conditional inclusion of files.  You can do the same thing for .go files instead of simple go tags.

Reply:
For reference about the file naming structure:  https://pkg.go.dev/cmd/go#hdr-Build_constraints

> If a file's name, after stripping the extension and a possible _test suffix, matches any of the following patterns:
> 
> *_GOOS
> *_GOARCH
> *_GOOS_GOARCH
> (example: source_windows_amd64.go) where GOOS and GOARCH represent any known operating system and architecture values respectively, then the file is considered to have an implicit build constraint requiring those terms (in addition to any explicit constraints in the file).

Since this isn't a .go file where we can use build tags, we use the file name instead.